apps/chrome-devtools/src/app-devtools/app.component.ts

```diff
@@ -27,32 +39,50 @@ import { ThemingPanelPresComponent } from './theming-panel/theming-panel-pres.co
     AppConnectionComponent,
     LocalizationPanelPresComponent,
     ThemingPanelPresComponent,
-    AsyncPipe
+    AsyncPipe,
+    StatePanelComponent,
+    FormsModule,
+    ReactiveFormsModule,
+    DfSelectModule,
+    DfTooltipModule,
+    JsonPipe
   ]
 })
-export class AppComponent implements OnDestroy {
-  private readonly subscription = new Subscription();
+export class AppComponent {
+  private readonly stateService = inject(StateService);
+  private readonly formBuilder = inject(FormBuilder);
+  private readonly connectionService = inject(ChromeExtensionConnectionService);
+  private readonly debugPanelService = inject(DebugPanelService);
+  private readonly rulesetHistoryService = inject(RulesetHistoryService);
 
-  public rulesetExecutions$: Observable<RulesetExecutionDebug[]>;
+  public readonly activeStateName = computed(() => this.stateService.activeState()?.name);
+  public readonly states = computed(() => Object.values(this.stateService.states()));
+  public readonly hasLocalChanges = this.stateService.hasLocalChanges;
+  public form = this.formBuilder.group({
+    activeStateName: new FormControl(this.activeStateName())
+  });
 
-  constructor(
-    connectionService: ChromeExtensionConnectionService,
-    debugPanelService: DebugPanelService,
-    rulesetHistoryService: RulesetHistoryService
-  ) {
-    this.rulesetExecutions$ = rulesetHistoryService.rulesetExecutions$;
+  public rulesetExecutions$ = this.rulesetHistoryService.rulesetExecutions$;
 
-    this.subscription.add(
-      connectionService.message$.subscribe((message) => {
-        if (isApplicationInformationMessage(message)) {
-          debugPanelService.update(message);
-        }
-      })
-    );
+  constructor() {
+    effect(() => {
+      this.form.controls.activeStateName.setValue(this.activeStateName(), { emitEvent: false });
+    });
+    const message = toSignal(this.connectionService.message$);
+    effect(() => {
+      const msg = message();
+      if (isApplicationInformationMessage(msg)) {
+        this.debugPanelService.update(msg);
+      }
+    });
+    const activateStateNameFormValueChanges = toSignal(this.form.controls.activeStateName.valueChanges, { initialValue: this.activeStateName() });
```

Comment:
It could be nice to establish some recommendation and consistency over the code on how we handle the link between a form valueChange and a signal